### PR TITLE
add recovered/vaccinated option to checkin

### PR DESCRIPTION
**Pull request by stringparser**
Ticket: https://github.com/railslove/recover-backlog/issues/179

In progress still. Adding you guys for review so you can check it or say if there something is missing. 

At the moment what I have to see is if we need something extra on the checkin page.

### CODE REVIEW

**Comment by stringparser** [comment]
Seems we already have a checkbox for confirming vaccination (see screenshot below). 

@juanmendi could you test here https://rcvr-app-9zukytc9o-railslove.vercel.app/ for the modal setup and let me know if that's enough? I've updated the locales and added the new option.

image

**Comment by bitboxer** [comment]
@stringparser did you check if the data exporter part shows a correct value with this?

**Comment by stringparser** [comment]
> @bitboxer did you check if the data exporter part shows a correct value with this?

I haven't worked yet with the exporter. Could you help with this?

File: lib/models/area.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    case 1:
+    case 24:
+    case 48: {
+      return CoronaTestOptions[value]
+    }
+    default: {
+      return null
+    }
+  }
+}
```
Comment (stringparser) [inline lib/models/area.ts:19]: tried `value in CoronaTestOptions` and `if (CoronaTestOptions[value])` but return value still showed `any`. This is what I could come up with without using typecasting.

**Comment by stringparser** [comment]
Ok, could make it work locally (see screenshot below were the Onboarding component doesn't show the provide test option radio button). 

@bitboxer not sure if we need some other adjustment. Let me know.

Here localhost checkin url for testing: 

http://localhost:3000/checkin?a=a4ad1d77-e900-4626-a785-8ae945bf2c7e&k=Ox9VVDPRqhUKO9hp%2FzZo%2BBiSF%2FG%2BbFfDGqWLntVWXl4%3D

Simply start the server as normal and as far as I can tell should work for you too.

image

File: lib/api/companies.ts [hunk 2]
```diff
@@ -23,10 +23,14 @@ export const CompanyTypeOptions = {
...
 
 export const CoronaTestOptions = {
-  0: 'kein Test notwendig',
-  24: 'maximal 24 Stunden alter Test notwendig',
-  48: 'maximal 48 Stunden alter Test notwendig',
-}
+  0: 'NO_TEST',
+  1: 'RECOVERED_OR_VACCINATED',
+  24: '24_HOUR_TEST_NEEDED',
+  48: '48_HOUR_TEST_NEEDED',
```
Comment (bitboxer) [inline lib/api/companies.ts:29]: Maybe it's time to rewrite this into propper enums?

**Comment by bitboxer** [comment]
@stringparser you just need to login as business owner and go to the export part. There you can request a new data dump and after that you can see the table there. I could show it to you via screen share if you want.

File: lib/api/companies.ts (hunk 2 above)
Comment (stringparser) [reply lib/api/companies.ts:29]: Is saved as `number` on the database. Actually tried to use `recovered` as key first but didn't work. So somehow we need the keys to be a number.

**Comment by stringparser** [comment]
> @stringparser you just need to login as business owner and go to the export part. There you can request a new data dump and after that you can see the table there. I could show it to you via screen share if you want.

ah I see. will check and if all looks good I'll merge

**Comment by stringparser** [comment]
> @stringparser you just need to login as business owner and go to the export part. There you can request a new data dump and after that you can see the table there. I could show it to you via screen share if you want.

can't figure it out. will merge since this was needed a while ago and check out with you separately. in case we need a fix we can deploy it separately.

**Pull request merged**